lib/Cake/Controller/Controller.php

```diff
@@ -329,7 +329,7 @@ public function __construct($request = null, $response = null) {
 		$this->Components = new ComponentCollection();
 
 		$childMethods = get_class_methods($this);
-		$parentMethods = get_class_methods('Controller');
+		$parentMethods = get_class_methods(get_parent_class($this));
```

Comment:
The intent here is to blacklist the methods that were originally declared in Controller e.g `redirect()` with this change, you will not be able to use inheritance to share action names.